### ISSUE TITLE
LMTP : Connection refused on all containers, emails not received both from the domain and from the outside

### ISSUE THREAD

**Issue by LeVraiRoiDHyrule** [Mailu/Mailu#2671]
### Environment

- [X] docker-compose
- [ ] kubernetes
- [ ] docker swarm

### Version

- Version: `1.9`

## Description

I am using mailu on my server with docker compose. The container names are not the based ones. They are named mailu-front, mailu-smtp, mailu-redis and so on. They are all on a dedicated docker network for mailu. The only container that have an access to the outside is the front. I am using env variables to tell mailu what are the names of the different containers, as follows:
```
ADMIN_ADDRESS=mailu-admin
ANTISPAM_ADDRESS=mailu-antispam
IMAP_ADDRESS=mailu-imap
REDIS_ADDRESS=mailu-redis
SMTP_ADDRESS=mailu-smtp
WEBMAIL_ADDRESS=mailu-webmail
LMTP_ADDRESS=mailu-imap
FRONT_ADDRESS=mailu-front
POP3_ADDRESS=mailu-imap
AUTHSMTP_ADDRESS=mailu-smtp
HOSTIMAP_ADDRESS=mailu-imap
```

I can perfectly send email from mailu. But mailu doesn't receive any email. Emails coming from the outside, **or between mailu addresses on the same domain**, all results in the following error:
```
2023-02-19T01:00:18.743121+01:00 abf7fa2de875 postfix/lmtp[400]: connect to mailu-imap[172.22.0.7]:24: Connection refused
2023-02-19T01:00:18.748272+01:00 abf7fa2de875 postfix/lmtp[400]: 59989BA2011: to=<admin@mydomain.fr>, relay=none, delay=1.9, delays=1.9/0.04/0/0, dsn=4.4.1, status=deferred (connect to mailu-imap[172.22.0.7]:24: Connection refused)
```

The problem seems to come the LMTP. Because if I change `LMTP_ADDRESS=mailu-imap` to `LMTP_ADDRESS=mailu-smtp`, the error becomes:
```
2023-02-19T01:00:18.743121+01:00 abf7fa2de875 postfix/lmtp[400]: connect to mailu-smtp[172.22.0.7]:24: Connection refused
2023-02-19T01:00:18.748272+01:00 abf7fa2de875 postfix/lmtp[400]: 59989BA2011: to=<admin@mydomain.fr>, relay=none, delay=1.9, delays=1.9/0.04/0/0, dsn=4.4.1, status=deferred (connect to mailu-smtp[172.22.0.7]:24: Connection refused)
```

So clearly there is a problem of dialogue between my containers. But I didn't find what it could be. I am stuck on this for months now. Any help, any answer is welcome. Have a great day.

## Replication Steps
Use the same configuration as me:
```yaml
  mailu-redis:
    image: redis:alpine
    container_name: mailu-redis
    restart: unless-stopped
    environment:
      - TZ=${TZ}
    networks:
      - mailu
    security_opt:
      - no-new-privileges:true
    volumes:
      - "${CONFIG_FOLDER}/mailu/redis:/data"
      #- /etc/localtime:/etc/localtime:ro
      #- /etc/timezone:/etc/timezone:ro
    depends_on:
      - mailu-resolver
    dns:
      - ${MAILU_NETWORK_IP}.254

  mailu-front:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}nginx:${MAILU_VERSION:-1.9}
    container_name: mailu-front
    restart: unless-stopped
    env_file: stack.env
    security_opt:
      - no-new-privileges:true
    environment:
      - TZ=${TZ}
    networks:
      - services
      - mailu
    #logging:
    #  driver: json-file
    ports:
    #  - "127.0.0.1:80:80"
    #  - "127.0.0.1:443:443"
      - "25:25"
      - "465:465"
      - "587:587"
      #- "110:110"
      #- "995:995"
      #- "143:143"
      - "993:993"
    volumes:
      - "${CONFIG_FOLDER}/caddy/data/caddy/certificates/acme-v02.api.letsencrypt.org-directory/mydomain.fr:/certs:ro"
      - "${CONFIG_FOLDER}/mailu/overrides/nginx:/overrides:ro"
      #- /etc/localtime:/etc/localtime:ro
      #- /etc/timezone:/etc/timezone:ro
    depends_on:
      - mailu-resolver
    dns:
      - ${MAILU_NETWORK_IP}.254
    labels:
      caddy: mail.mydomain.fr
      caddy.reverse_proxy: mailu-front:80

  mailu-resolver:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}unbound:${MAILU_VERSION:-1.9}
    container_name: mailu-resolver
    env_file: stack.env
    security_opt:
      - no-new-privileges:true
    environment:
      - TZ=${TZ}
    restart: unless-stopped
    networks:
      mailu:
        ipv4_address: ${MAILU_NETWORK_IP}.254

  mailu-admin:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}admin:${MAILU_VERSION:-1.9}
    container_name: mailu-admin
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true
    environment:
      - TZ=${TZ}
    networks:
      - mailu
    env_file: stack.env
    volumes:
      - "${CONFIG_FOLDER}/mailu/data:/data"
      - "${CONFIG_FOLDER}/mailu/dkim:/dkim"
      #- /etc/localtime:/etc/localtime:ro
      #- /etc/timezone:/etc/timezone:ro
    depends_on:
      - mailu-redis
      - mailu-resolver
    dns:
      - ${MAILU_NETWORK_IP}.254

  mailu-imap:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}dovecot:${MAILU_VERSION:-1.9}
    container_name: mailu-imap
    restart: unless-stopped
    env_file: stack.env
    security_opt:
      - no-new-privileges:true
    environment:
      - TZ=${TZ}
    networks:
      - mailu
    volumes:
      - "${CONFIG_FOLDER}/mailu/mail:/mail"
      - "${CONFIG_FOLDER}/mailu/overrides/dovecot:/overrides:ro"
      #- /etc/localtime:/etc/localtime:ro
      #- /etc/timezone:/etc/timezone:ro
    depends_on:
      - mailu-front
      - mailu-resolver
    dns:
      - ${MAILU_NETWORK_IP}.254

  mailu-smtp:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}postfix:${MAILU_VERSION:-1.9}
    container_name: mailu-smtp
    restart: unless-stopped
    env_file: stack.env
    security_opt:
      - no-new-privileges:true
    environment:
      - TZ=${TZ}
    networks:
      - mailu
      #- smtp
    volumes:
      - "${CONFIG_FOLDER}/mailu/mailqueue:/queue"
      - "${CONFIG_FOLDER}/mailu/overrides/postfix:/overrides:ro"
      #- /etc/localtime:/etc/localtime:ro
      #- /etc/timezone:/etc/timezone:ro
    depends_on:
      - mailu-front
      - mailu-resolver
    dns:
      - ${MAILU_NETWORK_IP}.254

  mailu-antispam:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}rspamd:${MAILU_VERSION:-1.9}
    container_name: mailu-antispam
    hostname: antispam
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true
    environment:
      - TZ=${TZ}
    networks:
      - mailu
    env_file: stack.env
    volumes:
      - "${CONFIG_FOLDER}/mailu/filter:/var/lib/rspamd"
      - "${CONFIG_FOLDER}/mailu/overrides/rspamd:/etc/rspamd/override.d:ro"
      #- /etc/localtime:/etc/localtime:ro
      #- /etc/timezone:/etc/timezone:ro
    depends_on:
      - mailu-front
      - mailu-resolver
    dns:
      - ${MAILU_NETWORK_IP}.254

  mailu-webmail:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}roundcube:${MAILU_VERSION:-1.9}
    container_name: mailu-webmail
    restart: unless-stopped
    env_file: stack.env
    security_opt:
      - no-new-privileges:true
    environment:
      - TZ=${TZ}
    networks:
      - mailu
    volumes:
      - "${CONFIG_FOLDER}/mailu/webmail:/data"
      - "${CONFIG_FOLDER}/mailu/overrides/roundcube:/overrides:ro"
      #- /etc/localtime:/etc/localtime:ro
      #- /etc/timezone:/etc/timezone:ro
    depends_on:
      - mailu-imap
      - mailu-resolver
    dns:
      - ${MAILU_NETWORK_IP}.254
```
Notes: "mail" TLS flavor is used, the front is under a caddy reverse proxy, all mailu containers are on a separated network, only the front has access to the outside

## Observed behaviour

An error occurs (see above)

## Expected behaviour

Mailu should be able to receive mails, both from the domain and from the outside

## Logs

See error above

**Comment by nextgens** [comment]
This is known to be broken on 1.9 but should have been fixed in master with #2562

Try to upgrade, don't forget to update your docker-compose.yml (that will have changed significantly) and re-open this issue if you can still reproduce it.

**Comment by nextgens** [comment]
To fix this specific issue on 1.9 all you need is:

```
HOST_LMTP='mailu-imap:2525'
```

But that's the visible part of the iceberg.

**Comment by LeVraiRoiDHyrule** [comment]
Thanks for your answer ! I upgraded to master and my problems are fixed now.
